shared/javascripts/local_storage.js

```diff
@@ -26,7 +26,7 @@ var ls = {
     if ( ls.localStorageDefined ) {
       localStorage.setItem(key, value);
     } else {
-      ls.preferences.setCharPref(key, value);
+      ls.storage[key] = value;
```

Comment:
if you could change this to ls.setItem(key, value), then they will more closely mirror each other. Assignment and function calls can do different things at times so it is better to eliminate that difference between then two branches of the conditionals.